### PR TITLE
Add tls assets for the clusterapi-operator

### PR DESCRIPTION
**Pull request by trawler**
This PR generates the certificate required by the clusterapi operator.
See this document, for more info: https://docs.google.com/document/d/1DuCROgrPBSTPrvjBWdxa3h6vi0dqKfTsQqhmrdCVqto/edit?ts=5b578ab8#heading=h.5j7iw1d1737j

### CODE REVIEW

File: installer/pkg/config-generator/tls.go [hunk 1]
```diff
@@ -253,6 +255,18 @@ func (c *ConfigGenerator) GenerateTLSConfig(clusterDir string) error {
...
 	}
+
+	// Cluster API cert
+	cfg = &tls.CertCfg{
+		Subject:   pkix.Name{CommonName: "cluster-apiserver", OrganizationalUnit: []string{"bootkube"}},
+		KeyUsages: x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		Validity:  validityTenYears,
+		IsCA:      true,
+	}
+	if _, _, err := generateCert(clusterDir, caKey, caCert, clusterAPIServerKeyPath, clusterAPIServerCertPath, cfg, false); err != nil {
```
Comment (wking) [inline installer/pkg/config-generator/tls.go]: I'm not familiar with the cluster API, so take this with a grain of salt, but are you sure you want `appendCA` (the last argument) `false` here?  I guess I don't understand why we'd want `appendCA` false for any CA besides the root one...

**Comment by wking** [comment]
`continuous-integration/jenkins/pr-merge` is reporting "This commit cannot be built" but [the details link][1] is giving me a 404.  Does anyone know what's going on there, or how we can get more information?

[1]: https://jenkins-tectonic-installer.prod.coreos.systems/job/openshift-installer/job/PR-79/4/display/redirect

**Comment by abhinavdahiya** [comment]
error from `continuous-integration/jenkins/pr-merge`
```
cluster_test.go:255: MISSING : [Kind: /v1:Secret | Namespace: default | Name: cluster-apiserver-certs] - [manifests/clusterapi-apiserver-secret.yaml]
smoke_test.go:112: failed with error: all defined resources were not present
```

**Comment by wking** [comment]
/lgtm cancel

While we sort out [the Jenkins error][1] and @abhinavdahiya's [name consistency points][2].

[1]: https://github.com/openshift/installer/pull/79#issuecomment-408471909
[2]: https://github.com/openshift/installer/pull/79#pullrequestreview-141181310

**Comment by wking** [comment]
[The aws-ec2 error][1] looks like a resource leak:

```
4 error(s) occurred:

* module.vpc.aws_elb.api_external: 1 error(s) occurred:

* aws_elb.api_external: DuplicateLoadBalancerName: Load Balancer named ci-op-1hr4m3pt-68485-ext already exists and it is configured with different parameters.
	status code: 400, request id: 459a072b-941b-11e8-80db-afef51e9359f
* module.vpc.aws_elb.tnc: 1 error(s) occurred:

* aws_elb.tnc: DuplicateLoadBalancerName: Load Balancer named ci-op-1hr4m3pt-68485-tnc already exists and it is configured with different parameters.
	status code: 400, request id: 452e4c60-941b-11e8-b015-ef27ac12059b
* module.vpc.aws_elb.console: 1 error(s) occurred:

* aws_elb.console: DuplicateLoadBalancerName: Load Balancer named ci-op-1hr4m3pt-68485-con already exists and it is configured with different parameters.
	status code: 400, request id: 4537e993-941b-11e8-9dff-5778efa1c445
* module.vpc.aws_elb.api_internal: 1 error(s) occurred:

* aws_elb.api_internal: DuplicateLoadBalancerName: Load Balancer named ci-op-1hr4m3pt-68485-int already exists and it is configured with different parameters.
	status code: 400, request id: 452d3aa1-941b-11e8-9536-0192182804fa
```

I'm running my stale-resource-prune scripts, and then I'll kick these tests again.

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/79/pull-ci-origin-installer-e2e-aws/290/build-log.txt

**Comment by wking** [comment]
All green with [the `aggregatorCA*` signing][1].

/lgtm

[1]: https://github.com/openshift/installer/pull/79#discussion_r206273904

**Comment by wking** [comment]
> openshift-merge-robot merged commit 4c1a82e into openshift:master

Did I [do that][1]?  I was expecting to only be able to add the lgtm label, because [I'm a reviewer and not an approver][2].  Is Prow ignoring `OWNERS` for some reason?

[1]: https://github.com/openshift/installer/pull/79#issuecomment-409001708
[2]: https://github.com/openshift/installer/blob/78e6f76292331a7fadc57ce3f362e558890fde84/OWNERS#L11

**Pull request merged**